src/reducers/trending.js

```diff
@@ -0,0 +1,32 @@
+import {
+  LOAD_TRENDING_REQUEST,
+  LOAD_TRENDING_SUCCESS
+} from '../actions/trending';
+
+export const defaultState = { loading: false, gifs: [] };
+
+const trending = (state = defaultState, { type, payload }) => {
+  switch (type) {
+    case LOAD_TRENDING_REQUEST: {
+      return {
+        ...state,
+        loading: true
+      };
+    }
+
+    case LOAD_TRENDING_SUCCESS: {
+      return {
+        ...state,
+        loading: false,
+        gifs: payload.data
+      };
+    }
```

Comment:
An optimistic programmer! ...But what about the FAIL case? 😅

Reply:
Same as above, did not add error handling due to lack of time 😅. Tho, If I'd probably refactor this to add error handling, and add a `requestize` saga helper, that makes a saga that does a request and dispatches actiopns for request, success and fail, as that behavior is fairly common